emu_mps/mps_backend_impl.py

```diff
@@ -165,6 +170,27 @@ def __setstate__(self, d: dict) -> None:
     def _get_autosave_filepath(autosave_prefix: str) -> pathlib.Path:
         return pathlib.Path(os.getcwd()) / (autosave_prefix + str(uuid.uuid1()) + ".dat")
 
+    def _optimise_interaction_matrix(
+            self,
+            f_mat: torch.Tensor,
+            m_mat: torch.Tensor,
+            optimise: bool = True
+            ) -> None:
+        if not optimise:
+            self.full_interaction_matrix = f_mat
+            self.masked_interaction_matrix = m_mat
+            return
+
+        f_mat_numpy = f_mat.numpy()
+        m_mat_numpy = m_mat.numpy()
```

Comment:
An idea is to always store in this class `opt_perm` and rename it to `qubit_permutation` which would default to the identity.
Then, you are only doing
`self.qubit_permutation = optmat.identity(self.num_sites) if not self._config.optimize_qubit_ordering else optimatrix.minimize_bandwidth(self.full_interaction_matrix)`

You would apply the identity permutation to stuff when optmat is disabled, but trust me that performance cost is very small while the code is nicer IMO